source/common/router/router.h

```diff
@@ -130,7 +130,7 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   void setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callbacks) override;
 
   // Upstream::LoadBalancerContext
-  Optional<uint64_t> hashKey() const override {
+  Optional<uint64_t> hashKey() override {
```

Comment:
Can you `ASSERT` at the top of here that the downstream cookies are empty? That would help catch multiple accidental calls.